userspace/graph-store/overview.md

```diff
@@ -0,0 +1,110 @@
++++
+title = "Graph Store Overview"
+weight = 1
+sort_by = "weight"
+template = "doc.html"
+insert_anchor_links = "right"
++++
+
+
+## Intro
+Graph store is a non-relational database suitable for use in building social media applications. It is primarily designed for storing text-based content and data that is generally threaded and nested. It is not suitable as a general purpose database for highly structured or pure binary data.
+
+<p align="center">
+  <img src="images/image1.png"/>
+</p>
+
+Graph Store is mainly in charge of two things: data storage and retrieval, and schema validation. There are two related tools: Graph Push Hook, which provides permissioning support to Graph Store and acts as a proxy layer to Graph Store for outside ships to access, and Graph Pull Hook, which can be used to request Graph Store data from other ships. Importantly, Graph Store itself does not have a built-in permissions system (or rather, is permissions-agnostic), and acts in a purely trusted manner, and assumes that all inputs are trusted This is why Graph Push Hook exists: to mediate untrusted requests from outside ships to your instance of Graph Store, rejecting invalid ones where a ship should not be able to modify the data.
```

Comment:
After "purely trusted" maybe we can say something about Graph Store *only allowing local connections to talk to it* and not allowing foreign ships at all.